UKHomeOffice/sgmr-service
# SPCR-443 Add VAT questions to pleasure craft form

### AC / Description of changes
Add the following questions to the pleasure craft form
- AIS
- MMSI
- VAT
- Charter or Hire


### To Test

- go to /pleasure-crafts
- create a new pc
- fill in page 1
- on page 2 click submit without completing any fields
- > all fields should error
- select AIS as no
- select MMSI as no
- select Charter or Hire as Neither of the above
- select VAT as no
- complete the other fields
- click submit
- > form should be submitted
- click the PC you just created to edit it
- go to page 2
- > AIS, MMSI, Charter or Hire, and VAT should all still show 'no'
- change them to yes (do not enter an MMSI number)
- click submit
- > MMSI field should error
- add an MMSI number
- click submit
- > form should be submitted

Go to voyage plans
- edit or create a voyage
- choose your new pleasure craft as the pc for that voyage
- go to check your answers page
- > you should see the new fields populated under the pleasure craft section based on what you saved earlier


### Notes

Currently selecting `neither of the above` for the `charter or hire` question will fail to submit as the API endpoint accepts only `Charter` or `Hire` or not having the field passed at all.

This is being investigated now

---

**Remember to merge to feature branches not `master`.**
